Alignment/TrackerAlignment/plugins/CosmicRateAnalyzer.cc

```diff
@@ -0,0 +1,532 @@
+   treeCluster	->Branch("DetID",&DetectorID);
+
+}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void 
+CosmicRateAnalyzer::endJob() 
+{
+//   file_->Write();
+//   file_->Close();
+}
+
+// ------------ method called when starting to processes a run  ------------
+void 
+CosmicRateAnalyzer::beginRun(edm::Run const&, edm::EventSetup const&)
+{
+   lastruntime		= 0.0;
+   lastrunnum		= 0.0;
+   ntrk_runnum 		= 0.0;
+   events		= 0.0;
+   track_BPIX           = 0.0;  
+   track_FPIX           = 0.0;
+   track_PIXEL          = 0.0;
+   track_TEC            = 0.0;
+   track_TECM           = 0.0;
+   track_TECP           = 0.0;
+   track_TOB            = 0.0;
+   track_TIB            = 0.0;
+   track_TID            = 0.0;
+   track_TIDM           = 0.0;
+   track_TIDP           = 0.0;
+}
+
+
+// ------------ method called when ending the processing of a run  ------------
+
+void 
+CosmicRateAnalyzer::endRun(edm::Run const&, edm::EventSetup const&)
+{
+//   if (lastrunnum !=0 && lastruntime !=0){
+//   std::cout<<"lastruntime : "<<lastruntime<<"       lastrunnum : "<<lastrunnum<<std::endl;
+//   std::cout<<" ################   : "<<ntrk_runnum<<std::endl;
+   number_of_tracks	=ntrk_runnum;
+   run_time		= lastruntime;
+   runnum		= lastrunnum;
+   number_of_tracks_PIX	= track_PIXEL ; 
+   number_of_tracks_FPIX= track_FPIX ; 
+   number_of_tracks_BPIX= track_BPIX; 
+   number_of_tracks_TEC	= track_TEC  ; 
+   number_of_tracks_TECM= track_TECM ; 
+   number_of_tracks_TECP= track_TECP ; 
+   number_of_tracks_TOB	= track_TOB  ; 
+   number_of_tracks_TIB	= track_TIB  ; 
+   number_of_tracks_TID	= track_TID  ; 
+   number_of_tracks_TIDM= track_TIDM ; 
+   number_of_tracks_TIDP= track_TIDP ; 
+   number_of_events	= events;
+   treeRun  ->Fill();
+}
+
+
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void 
+CosmicRateAnalyzer::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void 
+CosmicRateAnalyzer::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+CosmicRateAnalyzer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setComment("Create tuple with all variables required to calculate cosmic event and track rates.");
+  desc.add<edm::InputTag> ("tracksInputTag",edm::InputTag("ALCARECOTkAlCosmicsCTF0T"));
+  desc.add<edm::InputTag> ("muonsInputTag",edm::InputTag("muons1Leg"));
+  descriptions.addDefault(desc);
```

Comment:
@skeshri please change this line to
```
descriptions.add("cosmicRateAnalyzer", desc);
```
The first argument determines the name of your auto-generated config.